app/models/query/timestamps.rb

```diff
@@ -57,5 +60,15 @@ def timestamps=(array)
     def historic?
       timestamps != [Timestamp.now]
     end
+
+    private
+
+    def timestamps_are_parsable
+      invalid_timestamps = timestamps.reject(&:valid?)
+
+      if invalid_timestamps.any?
+        errors.add :timestamps, :invalid, values: invalid_timestamps.join(", ")
+      end
+    end
```

Comment:
The default for validations in the OP architecture is to have them in the contracts. The contracts will then be the place to check for a model's validity. This has the benefit of having different contracts for different situations (e.g. create vs update). That is not the case throughout the application, yet but we should work towards that.